code/datums/gamemode/dynamic/dynamic_rulesets_roundstart.dm

```diff
@@ -777,3 +759,146 @@ Assign your candidates in choose_candidates() instead.
+	for(var/mob/P in candidates)
+		if (P.mind.assigned_role && (P.mind.assigned_role in nanotrasen_staff))
+			assigned.Add(P)
+			candidates.Remove(P)
+
+/datum/dynamic_ruleset/roundstart/antag_madness/choose_candidates()
+	message_admins("DYNAMIC MODE: starting [name] with [assigned.len] heads of staff out of [candidates.len] players.")
+	log_admin("DYNAMIC MODE:  starting [name] with [assigned.len] heads of staff out of [candidates.len] players.")
+	return TRUE
+
+/datum/dynamic_ruleset/roundstart/antag_madness/ready(var/forced = 0)
+	return TRUE
+
+/datum/dynamic_ruleset/roundstart/antag_madness/execute()
+	antag_madness = TRUE
+	//first we initialize the nanotrasen faction, even if there are no heads currently there
+	var/datum/faction/nanotrasen/nanotrasen = find_active_faction_by_type(/datum/faction/nanotrasen)
+	if (!nanotrasen)
+		nanotrasen = ticker.mode.CreateFaction(/datum/faction/nanotrasen, null, 1)
+
+	//next we throw every head currently there in it
+	for(var/mob/M in assigned)
+		var/datum/role/nanotrasen_official/head = new
+		head.name = M.mind.assigned_role
+		head.AssignToRole(M.mind,1)
+		nanotrasen.HandleRecruitedRole(head)
+		head.Greet(GREET_ROUNDSTART)
+
+	//and now we manipulate Dynamic to force a bunch of other rulesets. It just works...
+	var/list/the_madness = list(
+		"Malfunctioning AI",
+		"Blob Conglomerate",
+		"Blood Cult",
+		"Nuclear Emergency",
+		"Wizard",
+		"Revolution",
+		"Vampires",
+		"Changelings",
+		"Syndicate Traitors",
+		)//The order matters. As a rule of thumb, rarer rulesets should have their chance come first, while more common ones will fill the spots
+
+	if (assigned.len < 5)
+		the_madness -= "Revolution"//No revs unless there's at least 5 roundstart heads, because that'd be overkill
+
+	for (var/madness_ruleset in the_madness)
+		for (var/datum/dynamic_ruleset/roundstart/rule in mode.roundstart_rules)
+			if (madness_ruleset == rule.name)
+				rule.candidates = candidates.Copy()
+				rule.trim_candidates()
+				if (rule.ready(TRUE))
+					rule.calledBy = "antag madness"
+					rule.required_candidates = 1//because we're funny
+					if (madness_ruleset == "Syndicate Traitors")//no traitor limits
+						rule.protected_from_jobs = list()
+						rule.restricted_from_jobs = list("Mobile MMI")
+						for (var/mob/M in candidates)
+							rule.assigned += M
+							rule.candidates -= M
+					else
+						rule.choose_candidates()
+					if (rule.assigned.len)
+						message_admins("ANTAG MADNESS: <font size='3'>[rule.name]</font> OK!")
+						log_admin("ANTAG MADNESS: <font size='3'>[rule.name]</font> OK!")
+						mode.threat_log += "[worldtime2text()]: Roundstart [rule.name] forced by antag madness"
+
+						if (istype(rule, /datum/dynamic_ruleset/roundstart/delayed/))
+							message_admins("ANTAG MADNESS: with a delay of [rule:delay/10] seconds.")
```

Comment:
Style policy calls for casting this as a delayed ruleset instead of using the trust operator